File: mirea/cources/2026-2027/aii-august/04/Part-02-Calculations-Modulo-a-Prime/examples/example-03.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
const ll mod = 1000000007;
ll power(ll a, ll exponent) {
	ll answ = 1;
	while (exponent > 0) {
		if (exponent % 2 == 1) answ = answ * a % mod;
		a = a * a % mod;
		exponent /= 2;
	}
	return answ;
}
main() {
	int n, k; cin >> n >> k;
	int limit = min(k, n - k);
	// Последовательно домножаем числитель и делим на очередной знаменатель.
	ll answ = 1;
	for (int i = 1; i <= limit; i++) {
		answ = answ * (n - i + 1) % mod;
		answ = answ * power(i, mod - 2) % mod;
	}
	cout << answ << '\n';
}

File: mirea/cources/2026-2027/aii-august/04/Part-02-Calculations-Modulo-a-Prime/examples/example-03.py
mod = 10 ** 9 + 7
n, k = map(int, input().split())
limit = min(k, n - k)
# Последовательно домножаем числитель и делим на очередной знаменатель.
answ = 1
for i in range(1, limit + 1):
	answ = answ * (n - i + 1) % mod
	answ = answ * pow(i, -1, mod) % mod
print(answ)